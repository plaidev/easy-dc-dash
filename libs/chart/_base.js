import d3 from 'd3'
import dc from 'dc'
import Store from '../store'
import {generateDomId, generateExtractor, reverseLegendOrder} from '../utils'
import {TIME_FORMATS, TIME_INTERVALS} from '../utils/time-format'
import ResetButton from './components/reset-button.vue'

export default {

  template: `<div class="krt-dc-component" :id="id">
              <reset-button v-on:reset="removeFilterAndRedrawChart()"></reset-button>
              <div v-text="title" style="font-size:24px; text-align:center;"></div>
            </div>`,

  components: {
    'reset-button': ResetButton
  },

  props: {
    dataset: {
      type: String,
      default: 'default'
    },
    dimension: {
    },
    reduce: {
    },
    id: {
      type: String,
      default: generateDomId
    },
    chartType: {
      type: String,
      default: 'barChart'
    },
    title: {
      type: String,
      default: ''
    },
    dateKey: {
      type: String
    },
    volume: {
      type: String
    },
    scale: {
      type: String
    },
    width: {
      type: Number
    },
    height: {
      type: Number
    },
    margins: {
      type: Object
    },
    xAxisLabel: {
      type: String,
      default: ''
    },
    xAxisFormat: {
      type: String,
      default: ''
    },
    yAxisLabel: {
      type: String,
      default: ''
    },
    yAxisFormat: {
      type: String,
      default: ''
    },
    renderLabel: {
      type: Boolean,
      default: true
    },
    renderTitle: {
      type: Boolean,
      default: true
    },
    useLegend: {
      type: Boolean,
      default: true
    },
    transitionDuration: {
      type: Number,
      default: 750
    },
    labels: {
    }
  },

  computed: {
    parent: function() {
      return `#${this.id}`;
    },
    dimensionName: function() {
      return this.dimension;
    },
    dimensionExtractor: function() {
      return generateExtractor(this.dimension)
    },
    reducerExtractor: function() {
      return generateExtractor(this.reduce)
    },
    grouping: function() {
      const grouping = this.dimensionExtractor;
      return Store.registerDimension(this.dimensionName, grouping, {dataset: this.dataset})
    },
    reducer: function() {
      const dim = Store.getDimension(this.dimensionName, {dataset: this.dataset});
      const reducer = this.reducerExtractor;
      return dim.group().reduceSum(reducer)
    },
    reduceKeys: function() {
      return []
    },
    accessor: function() {
      return null;
    },
    min: function() {
      const dim = this.grouping;
      const dimExtractor = this.dimensionExtractor;
      return dimExtractor(dim.bottom(1)[0]);
    },
    max: function() {
      const dim = this.grouping;
      const dimExtractor = this.dimensionExtractor;
      return dimExtractor(dim.top(1)[0]);
    },
    xScale: function() {
      let scale;
      if (!this.scale) return null;

      if (this.scale === 'time') scale = d3.time.scale;
      else scale = d3.scale[this.scale];

      if (!scale) return null;

      return scale().domain([this.min, this.max])
    }
  },

  methods: {
    removeFilterAndRedrawChart: function() {
      this.chart.filterAll();
      dc.redrawAll();
    },
    applyLegend: function(options={}) {
      if(!this.useLegend || !this.legend) return

      const {
        indexLabel = false,
        reverseOrder = false
      } = options;

      const l = this.legend
      
      const legendInstance = dc.legend()
        .x(l.x).y(l.y)
        .gap(l.gap).legendWidth(l.width)
        .itemWidth(l.itemWidth).itemHeight(l.itemHeight)
        .horizontal(l.horizontal)
        .legendText((d, i) => {
          const k = indexLabel? i: d.name;
          return this.getLabel(k)
        })

      this.chart.legend(legendInstance)
      if(reverseOrder) reverseLegendOrder(this.chart)
    },
    getLabel: function(key) {
      return Store.getLabel(key, {
        dataset: this.dataset,
        chartName: this.id
      })
    },
    getReduceKey: function(idx) {
      return this.reduceKeys && this.reduceKeys[idx] || idx
    getTimeInterval: function(key) {
      if((this.dateKey || this.timeScale) === undefined) return null
      else return TIME_INTERVALS[key]
    },
    getTimeFormat: function(key) {
      if((this.dateKey || this.timeScale) === undefined) return null
      else if (this.timeFormat) return d3.time.format(this.timeFormat)
      else return TIME_FORMATS[key]
    }
  },

  mounted: function() {
    const chart = Store.registerChart(
      this.parent,
      this.id,
      this.chartType,
      {volume: this.volume}
    );

    if (this.labels) {
      let labels = this.labels;
      if (typeof this.labels === 'string' || this.labels instanceof String) 
        labels = this.labels.split(',');
      Store.setLabels(labels, {
        dataset: this.dataset,
        chartName: this.id
      })
    }

    if (this.grouping) chart.dimension(this.grouping);
    if (this.reducer) {
      chart.group(this.reducer, this.getReduceKey(0));
    }
    if (this.accessor) chart.valueAccessor(this.accessor);
    if (this.xScale) chart.x(this.xScale);
    if (this.width) chart.width(this.width);
    if (this.height) chart.height(this.height);
    if (this.margins) chart.margins(this.margins);
    if (this.xAxisLabel) chart.xAxisLabel(this.xAxisLabel)
    if (this.yAxisLabel) chart.yAxisLabel(this.yAxisLabel)

    chart
      .renderLabel(this.renderLabel)
      .renderTitle(this.renderTitle)
      .transitionDuration(this.transitionDuration)
      .label((d) => {
        return this.getLabel(d.key)
      })
      .filterPrinter((filters) => {
        return filters
          .map((filter) => {
            return this.getLabel(dc.printers.filter(filter))
          })
          .join(', ')
      })

    this.chart = chart;

    return chart;
  },

  destroyed: function() {
    Store.unregisterChart(this.id);
    Store.unregisterDimension(this.dimensionName, {dataset: this.dataset})
  }
}
