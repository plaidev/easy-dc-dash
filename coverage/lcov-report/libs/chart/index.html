<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for libs/chart</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> libs/chart
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.94% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>164/391</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.03% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>109/330</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">45.05% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>41/91</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">45.09% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>147/326</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="_base.js"><a href="_base.js.html">_base.js</a></td>
	<td data-value="37.2" class="pic low"><div class="chart"><div class="cover-fill" style="width: 37%;"></div><div class="cover-empty" style="width:63%;"></div></div></td>
	<td data-value="37.2" class="pct low">37.2%</td>
	<td data-value="336" class="abs low">125/336</td>
	<td data-value="28.52" class="pct low">28.52%</td>
	<td data-value="277" class="abs low">79/277</td>
	<td data-value="39.24" class="pct low">39.24%</td>
	<td data-value="79" class="abs low">31/79</td>
	<td data-value="39.43" class="pct low">39.43%</td>
	<td data-value="279" class="abs low">110/279</td>
	</tr>

<tr>
	<td class="file medium" data-value="_coordinateGridBase.js"><a href="_coordinateGridBase.js.html">_coordinateGridBase.js</a></td>
	<td data-value="65.96" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 65%;"></div><div class="cover-empty" style="width:35%;"></div></div></td>
	<td data-value="65.96" class="pct medium">65.96%</td>
	<td data-value="47" class="abs medium">31/47</td>
	<td data-value="56.6" class="pct medium">56.6%</td>
	<td data-value="53" class="abs medium">30/53</td>
	<td data-value="81.82" class="pct high">81.82%</td>
	<td data-value="11" class="abs high">9/11</td>
	<td data-value="74.36" class="pct medium">74.36%</td>
	<td data-value="39" class="abs medium">29/39</td>
	</tr>

<tr>
	<td class="file high" data-value="ordinal-bar.vue"><a href="ordinal-bar.vue.html">ordinal-bar.vue</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 30 2017 17:42:44 GMT+0900 (JST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
