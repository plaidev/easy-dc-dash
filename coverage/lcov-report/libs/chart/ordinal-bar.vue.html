<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for libs/chart/ordinal-bar.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">libs/chart</a> ordinal-bar.vue
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&nbsp;
&lt;script lang='js'&gt;
&nbsp;
import d3 from "d3"
import dc from 'dc'
import Base from './_base'
import coordinateGridBase from './_coordinateGridBase'
import Store from '../store'
&nbsp;
export default {
  extends: coordinateGridBase,
&nbsp;
  props: {
    chartType: {
      type: String,
      default: 'barChart'
    },
    elasticX: {
      type: Boolean,
      default: true
    },
    elasticY: {
      type: Boolean,
      default: true
    },
    barPadding: {
      type: Number,
      default: 0.1
    },
    outerPadding: {
      type: Number,
      default: 0.5
    },
    removeEmptyRows: {
      default: true
    },
    useLegend: {
      default: false
    }
  },
  mounted: function() {
    const chart = this.chart;
    chart
      .barPadding(this.barPadding)
      .outerPadding(this.outerPadding)
      .x(d3.scale.ordinal())
      .xUnits(dc.units.ordinal)
      .elasticX(this.elasticX)
      .elasticY(this.elasticY)
      .renderVerticalGridLines(false)
    return chart
  }
}
&nbsp;
&lt;/script&gt;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 30 2017 17:42:44 GMT+0900 (JST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
