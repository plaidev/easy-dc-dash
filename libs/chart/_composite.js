import Vue from 'vue/dist/vue.js'
import Base from './_base'
import Store from '../store'
import {generateExtractor} from '../utils'
import {TIME_FORMATS} from '../utils/time-format'

function _getReduceKeySuper(Component) {
  if (!Component) return;
  if (Component.methods && Component.methods.getReduceKey)
    return Component.methods.getReduceKey
  return _getReduceKeySuper(Component.extends)
}


export function compose(Left, Right) {

  const ComponentObject = {
    extends: Base,

    template: `<div class="krt-dc-composite" :id="id">
                      <krt-dc-tooltip ref='tooltip'></krt-dc-tooltip>
                      <reset-button v-on:reset="removeFilterAndRedrawChart()"></reset-button>
                      <div v-text="title" style="font-size:24px; text-align:center;">{{title}}</div>
                    </div>`,

    props: {
      chartType: {
        type: String,
        default: 'compositeChart'
      },
      width: {
        type: Number,
        default: 240*4
      },
      height: {
        type: Number,
        default: 240
      },
      legend: {
        type: Object,
        default: () => {return {x:0, y:0, gap: 5, width: 800, itemWidth: 70, itemHeight: 12, horizontal: true}}
      },
      elasticY: {
        type: Boolean,
        default: true
      }
    },
    methods: {
      showTooltip: function(d, i) {
        const format = this.timeFormat ? this.timeFormat : d3.time.format('%Y-%m-%d');
        const fill = d3.event.target.getAttribute('fill');
        const stroke = d3.event.target.getAttribute('stroke');
        const color = fill || stroke;
        let key = null;
        let val = null;
        if (d.x && d.y) {
          key = this.scale ? format(d.x) : x;
          val = d.y;
        }
        else {
          key = this.getLabel(i);
        }
        const data = {
          key: key,
          val: val
        }
        this.$refs.tooltip.show(data, color)
      }
    },
    mounted: function() {

      // TODO: refactoring.

      const leftInstance = new Vue({
        extends: Left,
        computed: {
          parent: () => {
            return this.chart;
          },
          reducerExtractor: () => {
            return (d) => {
              const _reducer = generateExtractor(this.reduce);
              return _reducer(d)[0];
            }
          }
        },
        methods: {
          getReduceKey: function(idx) {
            const s = _getReduceKeySuper(Right);
            if (!s) return 'l'
            return 'l:' + s.apply(this, [idx])
          }
        },
        propsData: {
          dimension: this.dimension,
          scale: this.scale,
          legend: false
        }
      })

      const rightInstance = new Vue({
        extends: Right,
        computed: {
          parent: () => {
            return this.chart;
          },
          reducerExtractor: () => {
            return (d) => {
              const _reducer = generateExtractor(this.reduce);
              return _reducer(d)[1];
            }
          },
        },
        methods: {
          getReduceKey: function(idx) {
            const s = _getReduceKeySuper(Right);
            if (!s) return 'r'
            return 'r:' + s.apply(this, [idx])
          }
        },
        propsData: {
          dimension: this.dimension,
          scale: this.scale,
          legend: false
        }
      })

      Base.mounted.apply(leftInstance)
      Base.mounted.apply(rightInstance)

      const dim = this.grouping;
      const composite = this.chart;

      composite
        .dimension(dim)
        .compose([
          Left.mounted.apply(leftInstance),
          Right.mounted.apply(rightInstance).useRightYAxis(true),
        ])
        .renderHorizontalGridLines(true)
        .brushOn(false)
        //.rightY(scale.linear().domain([0, 1]))
        .elasticY(this.elasticY)
        .shareColors(true)

      // Compositeでまとめてlegendをつけるので、データ名について一貫した名前付けが必要
      // legendは配列として受け取り、番号で割り当てる
      // lines系なら問題ない。
      // TODO: bubbleチャート系だとうまくいかないかもしれない。
      // legendの利用有無も含めて再検討必要
      this.applyLegend({indexLabel: true})

      return composite.render();
    },

    destroyed: function() {
      // TODO: implement
      // Base.destroyed
    }
  }

  return ComponentObject;
}

