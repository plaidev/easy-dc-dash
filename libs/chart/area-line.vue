
<script lang='js'>

import StackedLines from './stacked-lines.vue'

export default {
  extends: StackedLines,
  props: {
    useDataPoints: {
      default: false
    }
  },
  methods: {
    _getValue: function(d, idx) {
      if (this.isRateReducer) {
        return d.value[idx].count !== 0 ? d.value[idx].value / d.value[idx].count : 0
      }
      return d.value[idx]
    },
    generateValueAccessor: function(idx) {
      return (d) => {
        if (idx === 0) return this._getValue(d, 0)
        else {
          return this._getValue(d, idx) - this._getValue(d, 0)
        }
      }
    }
  }
}
</script>

<style lang="less" module>
.chart-root :global {
  .dc-chart .stack._0 .area {
    fill-opacity: 0;
  }
}
</style>
