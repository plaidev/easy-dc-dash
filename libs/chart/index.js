import Base from './_base'
import {compose} from './_composite'
import SegmentPie from './segment-pie.vue'
import WeekRow from './week-row.vue'
import ListRow from './list-row.vue'
import RateLine from './rate-line.vue'
import StackedLines from './stacked-lines.vue'
import OrdinalBar from './ordinal-bar.vue'
import StackedBar from './stacked-bar.vue'
import FilterStackedBar from './filter-stacked-bar.vue'
import GeoJP from './geo-jp.vue'
import DataTable from './data-table.vue'
import resetAllButton from './reset-all-button.vue'

const components = {
  'segment-pie': SegmentPie,
  'week-row': WeekRow,
  'list-row': ListRow,
  'rate-line': RateLine,
  'stacked-lines': StackedLines,
  'ordinal-bar': OrdinalBar,
  'stacked-bar': StackedBar,
  'filter-stacked-bar': FilterStackedBar,
  'geo-jp': GeoJP,
  'data-table': DataTable,
  'stack-and-rate': compose(StackedLines, RateLine),
  'reset-all-button': resetAllButton
}

function install(Vue, options) {
  Vue.mixin({
    components: components
  })
}

export default {
  Base: Base,
  RateLine: RateLine,
  StackedLines: StackedLines,
  WeekRow: WeekRow,
  ListRow: ListRow,
  SegmentPie: SegmentPie,
  OrdinalBar: OrdinalBar,
  StackedBar: StackedBar,
  FilterStackedBar: FilterStackedBar,
  GeoJP: GeoJP,
  DataTable: DataTable,
  compose: compose,
  resetAllButton: resetAllButton,
  install: install,
  installedComponents: components
}
