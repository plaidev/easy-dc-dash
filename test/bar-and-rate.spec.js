import Vue from 'vue';
import { mount, createLocalVue } from 'vue-test-utils';
import CssModuleTestHelperMixin from './helpers/css-modules-test-helper-mixin';

import { compose } from '../libs/chart/_composite';
import Store from '../libs/store';

import OrdinalBar from '@/ordinal-bar.vue';
import RateLine from '@/rate-line.vue';

const BarAndRate = compose(OrdinalBar, RateLine);

const localVue = createLocalVue();
localVue.use(CssModuleTestHelperMixin);

describe('bar-and-rate', () => {
  Store.registerData([
    { date: new Date('2017-05-18 10:00'), d1: 'a', d2: true, v1: 1, v2: 10 },
    { date: new Date('2017-05-19 10:00'), d1: 'a', d2: false, v1: 2, v2: 20 },
    { date: new Date('2017-05-20 10:00'), d1: 'b', d2: false, v1: 3, v2: 30 },
    { date: new Date('2017-05-21 10:00'), d1: 'b', d2: true, v1: 5, v2: 50 }
  ]);

  const wrapper = mount(localVue.extend(BarAndRate), {
    localVue,
    propsData: {
      id: 'test-bar-and-rate',
      scale: 'time.day',
      dimension: 'd.date',
      reduce: '[[d.v1], {count: d.v1, value: d.v2}]'
    },
    attachToDocument: true
  });

  it('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
