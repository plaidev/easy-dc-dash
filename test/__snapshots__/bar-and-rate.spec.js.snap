// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bar-and-rate to match snapshot 1`] = `
<div class="outer-container" style="width: 377px; height: 233px;">
    <div class="backdrop"></div>
    <div class="card-container">
        <div class="inner-container" style="width: 377px; height: 233px;">
            <div class="container-header">
                <div class="icon-box">
                    <i class="fa fa-window-maximize"></i>
                </div>
                <!---->
            </div>
            <div class="render-area">
                <div id="test-bar-and-rate" class="krt-dc-component dc-chart" style="display: flex; position: relative; width: 100%; height: 100%;">
                    <!---->
                    <div class="reset-button">
                        <a class="reset" style="display: none;">
                            <i class="fa fa-ban">&nbsp;</i> <span>フィルター解除</span></a>
                    </div>
                    <!---->
                    <svg width="200" height="200">
                        <g>
                            <g class="grid-line horizontal" transform="translate(40,20)"></g>
                            <g class="grid-line vertical" transform="translate(40,20)">
                                <line x1="0" y1="150" x2="0" y2="0" opacity="0"></line>
                                <line x1="10" y1="150" x2="10" y2="0" opacity="0"></line>
                                <line x1="20" y1="150" x2="20" y2="0" opacity="0"></line>
                                <line x1="30" y1="150" x2="30" y2="0" opacity="0"></line>
                                <line x1="40" y1="150" x2="40" y2="0" opacity="0"></line>
                                <line x1="50" y1="150" x2="50" y2="0" opacity="0"></line>
                                <line x1="60" y1="150" x2="60" y2="0" opacity="0"></line>
                                <line x1="70" y1="150" x2="70" y2="0" opacity="0"></line>
                                <line x1="80" y1="150" x2="80" y2="0" opacity="0"></line>
                                <line x1="90" y1="150" x2="90" y2="0" opacity="0"></line>
                                <line x1="100" y1="150" x2="100" y2="0" opacity="0"></line>
                                <line x1="110" y1="150" x2="110" y2="0" opacity="0"></line>
                                <line x1="120" y1="150" x2="120" y2="0" opacity="0"></line>
                            </g>
                            <g class="chart-body" transform="translate(40, 20)" clip-path="url(about:blank#test-bar-and-rate-clip)"></g>
                            <g class="sub _0">
                                <g class="chart-body" transform="translate(40, 20)" clip-path="url(about:blank#test-bar-and-rate-clip)">
                                    <g class="stack _0">
                                        <rect class="bar" fill="#1f77b4" y="150" height="0"></rect>
                                        <rect class="bar" fill="#1f77b4" y="150" height="0"></rect>
                                        <rect class="bar" fill="#1f77b4" y="150" height="0"></rect>
                                        <rect class="bar" fill="#1f77b4" y="150" height="0"></rect>
                                        <text class="barLabel" text-anchor="middle"></text>
                                        <text class="barLabel" text-anchor="middle"></text>
                                        <text class="barLabel" text-anchor="middle"></text>
                                        <text class="barLabel" text-anchor="middle"></text>
                                    </g>
                                </g>
                            </g>
                            <g class="sub _1">
                                <g class="chart-body" transform="translate(40, 20)" clip-path="url(about:blank#test-bar-and-rate-clip)">
                                    <g class="stack-list">
                                        <g class="stack _0">
                                            <path class="line" stroke="#ff7f0e"></path>
                                            <text class="lineLabel" text-anchor="middle"></text>
                                            <text class="lineLabel" text-anchor="middle"></text>
                                            <text class="lineLabel" text-anchor="middle"></text>
                                            <text class="lineLabel" text-anchor="middle"></text>
                                        </g>
                                    </g>
                                    <g class="dc-tooltip-list">
                                        <g class="dc-tooltip _0">
                                            <path class="yRef" style="display: none;" stroke-dasharray="5,5"></path>
                                            <path class="xRef" style="display: none;" stroke-dasharray="5,5"></path>
                                            <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#ff7f0e"></circle>
                                            <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#ff7f0e"></circle>
                                            <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#ff7f0e"></circle>
                                            <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#ff7f0e"></circle>
                                        </g>
                                    </g>
                                </g>
                            </g>
                            <g class="axis x" transform="translate(40,170)">
                                <g class="tick" style="opacity: 0.000001;" transform="translate(0,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Thu 18</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(10,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">06 AM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(20,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">12 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(30,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">06 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(40,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Fri 19</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(50,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">06 AM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(60,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">12 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(70,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">06 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(80,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Sat 20</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(90,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">06 AM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(100,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">12 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(110,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">06 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(120,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">May 21</text>
                                </g>
                                <path class="domain"></path>
                            </g>
                            <g class="axis y" transform="translate(40,20)">
                                <path class="domain"></path>
                            </g>
                            <g class="axis yr" transform="translate(160,20)">
                                <path class="domain"></path>
                            </g>
                        </g>
                        <defs>
                            <clippath id="test-bar-and-rate-clip">
                                <rect width="120" height="150" transform="translate(-0, -0)"></rect>
                            </clippath>
                        </defs>
                    </svg>
                </div>
            </div>
        </div>
    </div>
</div>
`;
